lib/src/Rating/SVGComponent.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from "react";
+
+export type SVGProps = JSX.IntrinsicElements["div"] & {
+    colors?: [string, string];
+    width: React.ReactText;
+    height: React.ReactText;
+    title?: string;
+    index?: number;
+    value?: number;
+    step?: number;
+    customSVG?: any;
+};
+
+export const SVGComponent: React.FC<SVGProps> = ({ colors, width, height, title, index, value, customSVG, step }: SVGProps) => {
+    const [activePercentage, setActivePercentage] = useState(value);
+
+    useEffect(() => {
+        if (value >= index) {
+            setActivePercentage((value - index) * 100);
+        }
+    }, [value]);
+
+    const getGradientId = (defaultValue?: string): string => {
+        switch (step) {
+            case 1:
+                return "url(#full_grad)";
+            case 0.5:
+                return defaultValue || "url(#half_grad)";
+            default:
+                return `url(#dynamic_grad${index})`;
+        }
+    };
```

Comment:
Since function can be cached since it doesn't have a lot of changing dependencies
```suggestion
    const getGradientId = React.useCallback((defaultValue?: string): string => {
        switch (step) {
            case 1:
                return "url(#full_grad)";
            case 0.5:
                return defaultValue || "url(#half_grad)";
            default:
                return `url(#dynamic_grad${index})`;
        }
    }, [step, index]);
```

Reply:
done